sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/HardcodedIpCheck.kt

```diff
@@ -85,10 +86,14 @@ private fun isValidIPV6(ipv6: String, ipv4: String?): Boolean {
 }
 
 private fun isIPV4Exception(ip: String) = ip.startsWith("127.")
-        || "255.255.255.255" == ip || "0.0.0.0" == ip || ip.startsWith("2.5.") || isReservedDocumentationIp(ip)
+        || "255.255.255.255" == ip || "0.0.0.0" == ip || ip.startsWith("2.5.")
+        || isReservedDocumentationIp(ip) || isLocalIPv4MappedIPv6(ip)
 
 private fun isIPV6Exception(ip: String) = IPV6_LOOPBACK.matches(ip) || IPV6_NON_ROUTABLE.matches(ip) || isReservedDocumentationIp(ip)
+        || isLocalIPv4MappedIPv6(ip)
 
 private fun getCompressionSeparatorCount(str: String) = str.split("::").size - 1
 
-private fun isReservedDocumentationIp(str: String) : Boolean = RESERVED_DOC_IP_RANGE.any { it -> str.startsWith(it) }
\ No newline at end of file
+private fun isReservedDocumentationIp(str: String) : Boolean = RESERVED_DOC_IP_RANGE.any { it -> str.startsWith(it) }
+
+private fun isLocalIPv4MappedIPv6(str: String) : Boolean = LOCAL_IPV4_MAPPED_IPV6.any { it -> str.startsWith(it, true) }
```

Comment:
When you write functions that return a single expression, like here, you can actually omit the return type declaration entirely. So 
```
private fun isReservedDocumentationIp(str: String) : Boolean = RESERVED_DOC_IP_RANGE.any { it -> str.startsWith(it) }
```
would become 
```
private fun isReservedDocumentationIp(str: String) = RESERVED_DOC_IP_RANGE.any { it -> str.startsWith(it) }
```
This makes refactoring much simpler, as types aren't declared redundantly. Note that if you haven't done so already, it helps to enable type hints in the IDE and you'll still be able to see the type while editing the code.